### PR TITLE
fix: adds delivery fees config and base weights

### PR DESCRIPTION
**Pull request by al3mart**
Adds a default configuration for delivery fees that doesn't charge to the relay nor the sibling system parachains and collecting the fee in the account corresponding to the treasury module (non existent in the template right now) .
Also declares the type `WeightInfo` for each pallet with the base weights provided in the pallet.

Motivation for changes comes from similarities found at SRLabs audit: https://github.com/OpenZeppelin/polkadot-generic-runtime-template/blob/main/audits/2024-04.pdf

### CODE REVIEW

**Comment by evilrobot-01** [comment]
Nice! Can we add some unit tests in the runtime to assert that these values are set correctly, and remain that way so that no regressions occur?

**Comment by evilrobot-01** [comment]
Can we reorder https://github.com/r0gue-io/base-parachain/blob/9a312bbf174e1d5a2780000cfb7b178ad6c4bd52/runtime/src/lib.rs#L539 in line with the order of the pallets defined in the runtime, ensuring that all those that are included in the runtime are included in this list?

It would be good to also add a comment somewhere within construct_runtime as a reminder to ensure that they are added to define_benchmarks, for any template users.

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -311,7 +311,7 @@ impl pallet_timestamp::Config for Runtime {
 	type MinimumPeriod = ConstU64<0>;
 	#[cfg(not(feature = "experimental"))]
 	type MinimumPeriod = ConstU64<{ SLOT_DURATION / 2 }>;
-	type WeightInfo = ();
+	type WeightInfo = pallet_timestamp::weights::SubstrateWeight<Runtime>;
```
Comment (evilrobot-01) [inline runtime/src/lib.rs:314]: Whilst this is included in the audit report, I am not sure if this is strictly necessary as the weights defined in the pallet are implemented for the unit type (https://github.com/paritytech/polkadot-sdk/blob/c973fe86f8c668462186c95655a58fda04508e9a/substrate/frame/timestamp/src/weights.rs#L84), exactly the same as the implementation used here. It is more explicit however.

I do agree that these should be replaced with actual benchmarks based on this specific runtime configuration on reference hardware, which is not the case when using the weights from the frame pallets as that benchmarked runtime _may_ have a different config to this one.

Comment (al3mart) [reply runtime/src/lib.rs:314]: Ah good catch, I missed this.

Same story with `pallet_sudo` - https://github.com/paritytech/polkadot-sdk/blob/c973fe86f8c668462186c95655a58fda04508e9a/substrate/frame/sudo/src/weights.rs#L108

Comment (al3mart) [reply runtime/src/lib.rs:314]: Every `WeigthInfo` implementation for each pallet has at the moment of writing the same weights defined than those in their `SubstrateWeight` implementation.

While the implementation for `()` is meant to exist for testing and the one provided in `SubstrateWeight` is meant to follow Polkadot's hw requirements, they seem to be updated together.

`()` does a better job making the developer aware that `WeightInfo` should be updated with their benchmark results.

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -485,12 +509,115 @@ impl pallet_collator_selection::Config for Runtime {
...
+	type WeightInfo = (); // Configure based on benchmarking results.
+}
+
+parameter_types! {
+	pub const NftFractionalizationPalletId: PalletId = PalletId(*b"fraction");
+	pub NewAssetSymbol: BoundedVec<u8, AssetsStringLimit> = (*b"FRAC").to_vec().try_into().unwrap();
+	pub NewAssetName: BoundedVec<u8, AssetsStringLimit> = (*b"Frac").to_vec().try_into().unwrap();
+}
+
+impl pallet_nft_fractionalization::Config for Runtime {
```
Comment (evilrobot-01) [inline runtime/src/lib.rs:521]: Why is this here?!

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -485,12 +509,115 @@ impl pallet_collator_selection::Config for Runtime {
...
+	type StringLimit = AssetsStringLimit;
+	type Freezer = ();
+	type Extra = ();
+	type WeightInfo = (); // Configure based on benchmarking results.
+	type CallbackHandle = ();
+	type AssetAccountDeposit = AssetAccountDeposit;
+	type RemoveItemsLimit = ConstU32<1000>;
+	#[cfg(feature = "runtime-benchmarks")]
+	type BenchmarkHelper = ();
+}
```
Comment (weezy20) [inline runtime/src/lib.rs:615]: Why is pallet-nfts and nft-fractionalization a part of the base-parachain?

**Review by evilrobot-01 — APPROVED** [review]
Made a few tweaks, so looks good to me. 

Ideally we would have tests for these changes, but that might be overkill for a template. We can always port any applicable tests built downstream back here in the future.

**Review by Daanvdplas — APPROVED** [review]

**Review by peterwht — APPROVED** [review]
